src/cascadia/TerminalSettingsModel/FileUtils.cpp

```diff
@@ -39,87 +43,60 @@ namespace winrt::Microsoft::Terminal::Settings::Model
         return baseSettingsPath;
     }
 
-    locked_hfile OpenFileReadSharedLocked(const std::filesystem::path& path)
+    // Function Description:
+    // - Checks the permissions on this file, to make sure it can only be opened
+    //   for writing by admins. We will be checking to see if the file is owned
+    //   by the Builtin\Administrators group. If it's not, then it was likely
+    //   tampered with.
+    // Arguments:
+    // - path: the path to the file to check
+    // Return Value:
+    // - true if it had the expected permissions. False otherwise.
+    static bool _hasElevatedOnlyPermissions(const std::filesystem::path& path)
     {
-        wil::unique_hfile file{ CreateFileW(path.c_str(), GENERIC_READ, FILE_SHARE_READ | FILE_SHARE_WRITE, nullptr, OPEN_EXISTING, FILE_ATTRIBUTE_NORMAL, nullptr) };
-        THROW_LAST_ERROR_IF(!file);
-        // just lock the entire file
-        OVERLAPPED sOverlapped;
-        sOverlapped.Offset = 0;
-        sOverlapped.OffsetHigh = 0;
-        // Shared lock
-        THROW_LAST_ERROR_IF(!LockFileEx(file.get(),
-                                        0, // lock shared, wait to return until lock is obtained
-                                        0, // reserved, does nothing
-                                        INT_MAX, // lock INT_MAX bytes
-                                        0, // higher-order bytes, if our state file is greater than 2GB I guess this will be a problem
-                                        &sOverlapped));
-        return { std::move(file), sOverlapped };
+        // If the file is owned by the administrators group, trust the
+        // administrators instead of checking the DACL permissions. It's simpler
+        // and more flexible.

```

Comment:
I was thinking the name and description would change to something like `_isOwnedByAdministrators()`. The flexibility with this approach is that an administrator can grant write access to standard users by lowering the integrity level from high to medium via `icacls.exe filename /setintegritylevel M`, and also grant write access to specific users as needed (e.g. a service/task virtual SID, "LOCAL SERVICE", etc). When the file is overwritten (as opposed to deleted), the filesystem retains the existing security descriptor, so the owner doesn't change.

Reply:
![the best kind](https://c.tenor.com/fVtcUsX--ZsAAAAC/correct-futurama.gif)